File: examples/stream/createChannel/go/createChannel.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/stream"
)

// Create a channel with a given name and description. Channels are created automatically but
// this allows you to specify a description that's persisted for the lifetime of the channel.
func CreateChannel() {
	streamService := stream.NewStreamService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := streamService.CreateChannel(&stream.CreateChannelRequest{
		Description: "The channel for all things",
		Name:        "general",
	})
	fmt.Println(rsp, err)
}

File: examples/stream/createChannel/node/createChannel.js
const { StreamService } = require("micro-js-client/stream");

// Create a channel with a given name and description. Channels are created automatically but
// this allows you to specify a description that's persisted for the lifetime of the channel.
async function createChannel() {
  let streamService = new StreamService(process.env.MICRO_API_TOKEN);
  let rsp = await streamService.createChannel({
    description: "The channel for all things",
    name: "general",
  });
  console.log(rsp);
}

createChannel();
